en/components/grid.md

```diff
@@ -73,14 +73,81 @@ The Grid comes with styling flexibility achievable through styling the individua
 
-| Do                          | Don't                         |
-| --------------------------- | ----------------------------- |
+| Do                                                                         | Don't                                                                          |
+| -------------------------------------------------------------------------- | ------------------------------------------------------------------------------ |
 | <img src="../images/grid_do1.png" srcset="../images/grid_do1@2x.png 2x" /> | <img src="../images/grid_dont1.png" srcset="../images/grid_dont1@2x.png 2x" /> |
 
 ## Code generation
 
-`🕹️DataSource`
-`🕹️Event`
+When colors or fonts are specified for the Grid, the Grid HTML element will be wrapped in a div. This is required by browsers to style a nested component (a component within another component). The grid has to be a detached symbol to be able to be used. The width of the grid is determined be adding up the widths of the columns. And the height of the grid is set to null so as to render rows properly even if the container for the grid doesn’t have a height set.
+
+### Data Bindings
+
+Data bindings are specified by using curly brace syntax, example: {isAdmin}. Text fields (not DataProperty or DataSource) also support string interpolation syntax example: Admin: {isAdmin}. Data bindings can be non-nested or nested. If the target property is a nested property, include the nested property chain, but don’t include the model object name. Examples:
+
+#### Not Nested
+
+```typescript
+Customer {
+  imageName: String;
+}
+```
+
+DataProperty would be: `{imageName}`
+
+#### Nested
+
+```typescript
+Profile {
+  imageName: String;
+}
+
+Customer {
+  profile: Profile;
+}
+```
+
+DataProperty would be: `{profile.imageName}`
+
+### DataSource
+
+When supplied, the `🕹️DataSource` value is used to set up the data source for the grid and needs to be an array of objects, where each column name is a property on the object. If the data source is not supplied the grid will only render the column headers.
+
+### Header
+
+When supplied, the Headers help determine the number of columns to be rendered. It will try and pair a Header with a Body(Cell) that is below it to form the column. If no matching Body can be found a column will be created based on the information that can be gathered from the Header. When a pair is found information will be gathered from the Header first then the Body/Cell.
+
+#### Text
+
+The Header Text property may contain text, binding, or a combination of the two, examples:
+
+- Settings
+- {settingsLabel}
+- Important {labelText}
+
+#### Sorting
+
+When supplied enables sorting on the specified column.
+
+#### Filter State
+
+When supplied enables the sorting on the specified column.
```

Comment:
Should this enable filtering?  And "the" is unnecessary here.  "When supplied enables filtering on the specified column."